modin/engines/base/io/text/csv_reader.py

```diff
@@ -67,13 +68,20 @@ def _read(cls, filepath_or_buffer, **kwargs):
                 filepath_or_buffer,
                 **dict(kwargs, usecols=None, nrows=0, skipfooter=0, index_col=None),
             ).columns
+        elif index_col is None and not usecols:
+            empty_pd_df = pandas.read_csv(filepath_or_buffer, nrows=0)
```

Comment:
A brief comment here on why this is needed/what the resulting behavior is would be really helpful.
```suggestion
            # When names is set to some list that is smaller than the number of columns
            # in the file, the first columns are built as a hierarchical index.
            empty_pd_df = pandas.read_csv(filepath_or_buffer, nrows=0)
```

Reply:
Comments added.